zos-utils/zossrc/javaUsermap.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+#define _OPEN_SYS
+#include <unistd.h>
+#include <errno.h>
+#include "zowe-common-c/h/rusermap.h"
+#include "javaUsermap.h"
+#include <stdio.h>
+/**
+ * Define version of JNI for this library
+ */
+#define JNI_VERSION JNI_VERSION_1_8
+
+#pragma convert(819)
+const char *JNI_CLASS_ILLEGAL_ARGUMENT_EXCEPTION = "java/lang/IllegalArgumentException";
+
+const char *JNI_SIGNATURE_METHOD_STRING_INT_INT_INT_INT_VOID = "(Ljava/lang/String;IIII)V";
+const char *JNI_SIGNATURE_METHOD_STRING_INT_INT_INT_VOID = "(Ljava/lang/String;III)V";
+const char *JNI_METHOD_CONSTRUCTOR = "<init>";
+
+const char *JNI_CLASS_MAPPER_RESPONSE = "org/zowe/commons/usermap/MapperResponse";
+const char *JNI_CLASS_CERTIFICATE_RESPONSE = "org/zowe/commons/usermap/CertificateResponse";
+
+const char *JNI_MESSAGE_CANNOT_CONVERT_USER_ID = "Cannot convert userID";
+const char *JNI_MESSAGE_DN_NAME_TOO_LONG = "Distinguished name is not allowed to be more than 246 characters";
+const char *JNI_MESSAGE_REGISTRY_NAME_TOO_LONG = "Registry name is not allowed to be more than 255 characters";
+#pragma convert(0)
+
+jclass certificateClass;
+jmethodID certificateClassCtor;
+jclass mapperClass;
+jmethodID mapperClassCtor;
+jclass exception_clazz;
+
+int strnlen(char *txt, int max) {
+    if (max < 0) return 0;
+    for (int i = 0; i < max; i++) {
+        if (!txt[i]) return i;
+    }
+    return max;
+}
+
+/**
+ * Return Java environment by virtual machine. It is useful for load and unload event.
+ */
+JNIEnv* getEnv(JavaVM *vm) {
+    JNIEnv* env;
+    if ((*vm)->GetEnv(vm, (void **)&env, JNI_VERSION) != JNI_OK) {
+        return NULL;
+    }
+    return env;
+}
+
+ jint JNI_OnLoad(JavaVM *vm, void *reserved) {
+     JNIEnv* env = getEnv(vm);
+     if (env == NULL) return JNI_ERR;
+     certificateClass = (*env) -> NewGlobalRef(env, (*env) -> FindClass(env, JNI_CLASS_CERTIFICATE_RESPONSE));
+     certificateClassCtor = (*env) -> GetMethodID(env, certificateClass, JNI_METHOD_CONSTRUCTOR, JNI_SIGNATURE_METHOD_STRING_INT_INT_INT_VOID);
+     mapperClass = (*env) -> NewGlobalRef(env, (*env) -> FindClass(env, JNI_CLASS_MAPPER_RESPONSE));
+
+     mapperClassCtor = (*env) -> GetMethodID(env, mapperClass, JNI_METHOD_CONSTRUCTOR, JNI_SIGNATURE_METHOD_STRING_INT_INT_INT_INT_VOID);
+
+     exception_clazz = (*env) -> NewGlobalRef(env, (*env) -> FindClass(env, JNI_CLASS_ILLEGAL_ARGUMENT_EXCEPTION));
```

Comment:
```suggestion
     certificateClass = (*env) -> NewGlobalRef(env, (*env) -> FindClass(env, JNI_CLASS_CERTIFICATE_RESPONSE));
     if (certificateClass == NULL) return JNI_ERR;
     certificateClassCtor = (*env) -> GetMethodID(env, certificateClass, JNI_METHOD_CONSTRUCTOR, JNI_SIGNATURE_METHOD_STRING_INT_INT_INT_VOID);
     if (certificateClassCtor == NULL) return JNI_ERR;
     mapperClass = (*env) -> NewGlobalRef(env, (*env) -> FindClass(env, JNI_CLASS_MAPPER_RESPONSE));
     if (mapperClass == NULL) return JNI_ERR;

     mapperClassCtor = (*env) -> GetMethodID(env, mapperClass, JNI_METHOD_CONSTRUCTOR, JNI_SIGNATURE_METHOD_STRING_INT_INT_INT_INT_VOID);
     if (mapperClassCtor== NULL) return JNI_ERR;

     exception_clazz = (*env) -> NewGlobalRef(env, (*env) -> FindClass(env, JNI_CLASS_ILLEGAL_ARGUMENT_EXCEPTION));
     if (exception_clazz == NULL) return JNI_ERR;
```

To avoid throwing an exception while another was thrown.